pkg/controller/sriovnetwork/sriovnetwork_controller.go

```diff
@@ -141,33 +143,66 @@ func (r *ReconcileSriovNetwork) Reconcile(request reconcile.Request) (reconcile.
 	if err := controllerutil.SetControllerReference(instance, netAttDef, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
-
-	// Check if this NetworkAttachmentDefinition already exists
-	found := &netattdefv1.NetworkAttachmentDefinition{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: netAttDef.Name, Namespace: netAttDef.Namespace}, found)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
-			err = r.client.Create(context.TODO(), netAttDef)
+	if lastNetworkNamespace == "" {
+		lastNetworkNamespace = netAttDef.GetNamespace()
+	}
+	if lastNetworkNamespace == netAttDef.GetNamespace() {
+		// networkNamespace not updated
+		namespaces = netAttDef.GetNamespace()
+		// Check if this NetworkAttachmentDefinition already exists
+		found := &netattdefv1.NetworkAttachmentDefinition{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: netAttDef.Name, Namespace: namespaces}, found)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
+				err = r.client.Create(context.TODO(), netAttDef)
+				if err != nil {
+					reqLogger.Error(err, "Couldn't create NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
+					return reconcile.Result{}, err
+				}
+				lastNetworkNamespace = netAttDef.GetNamespace()
+			}
+			// NetworkAttachmentDefinition created successfully - don't requeue
+			return reconcile.Result{}, err
+		}
+		if lastNetworkNamespace != netAttDef.GetNamespace() {
```

Comment:
Will this condition ever be `true`? It's in the loop of `lastNetworkNamespace == netAttDef.GetNamespace()`;  if netAttDef is created in the loop,  `lastNetworkNamespace` will be assigned with `netAttDef.GetNamespace()`.

Reply:
Fixed.